core/src/main/kotlin/org/eln2/sim/electrical/mna/Circuit.kt

```diff
@@ -341,10 +339,8 @@ class Circuit {
         componentsChanged = true
         // This is the ONLY place where this should be set.
         comp.circuit = this
-        for (i in 0 until comp.nodeCount) {
-            val n = Node(this)
-            compNodeMap[n] = comp
-            comp.nodes.add(n)
+        for (i in 0 until comp.pinCount) {
```

Comment:
Replace with `(0 .. until comp.pinCount).forEach{}` style call.

Reply:
See the performance lint above.